File: src/main/java/no/nav/veilarbaktivitet/aktivitetskort/dto/kassering/KasseringsBestilling.java
package no.nav.veilarbaktivitet.aktivitetskort.dto.kassering;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import no.nav.common.types.identer.NavIdent;
import no.nav.common.types.identer.NorskIdent;
import no.nav.veilarbaktivitet.aktivitetskort.ActionType;
import no.nav.veilarbaktivitet.aktivitetskort.dto.BestillingBase;

import java.util.UUID;

@SuperBuilder(toBuilder = true)
@Getter
@NoArgsConstructor
public class KasseringsBestilling extends BestillingBase {
    private NavIdent navIdent;
    private NorskIdent personIdent;
    private UUID aktivitetsId;
    private String begrunnelse;

    public KasseringsBestilling(String source, UUID messageId, ActionType actionType, NavIdent navIdent, NorskIdent personIdent, UUID aktivitetsId, String begrunnelse) {
        super(source, messageId, actionType);
        this.navIdent = navIdent;
        this.personIdent = personIdent;
        this.aktivitetsId = aktivitetsId;
        this.begrunnelse = begrunnelse;
    }

    @Override
    public UUID getAktivitetskortId() {
        return this.aktivitetsId;
    }
}

File: src/main/kotlin/no/nav/veilarbaktivitet/aktivitetskort/bestilling/KasseringsBestilling.kt
package no.nav.veilarbaktivitet.aktivitetskort.bestilling

import no.nav.common.types.identer.NavIdent
import no.nav.common.types.identer.NorskIdent
import no.nav.veilarbaktivitet.aktivitetskort.ActionType
import no.nav.veilarbaktivitet.aktivitetskort.dto.BestillingBase
import java.util.*

class KasseringsBestilling(
    source: String?,
    messageId: UUID?,
    actionType: ActionType?,
    val navIdent: NavIdent,
    val personIdent: NorskIdent,
    val aktivitetsId: UUID,
    val begrunnelse: String?
) : BestillingBase(source, messageId, actionType) {
    override fun getAktivitetskortId(): UUID {
        return aktivitetsId
    }
}
